ccm_web/server/src/lti/lti.service.ts

```diff
@@ -45,7 +47,31 @@ export class LTIService implements BeforeApplicationShutdown {
 
     // Redirect to the application root after a successful launch
     provider.onConnect(async (token: IdToken, req: Request, res: Response) => {
-      logger.debug(JSON.stringify(token.userInfo))
+      logger.debug(`LTI launch is successfull! ${JSON.stringify(token.userInfo)}`)
+      const customLTIVariables = token.platformContext.custom
+      if (customLTIVariables == null) {
+        return res.json({ lti_error: 'LTI launch is missing custom attributes, please add it in LTI configutation in Canvas' })
+      }
+      const loginId = customLTIVariables.login_id as string
+      const [record, created] = await this.userService.upsertUser({
+        firstName: token.userInfo.given_name,
+        lastName: token.userInfo.family_name,
+        email: token.userInfo.email,
+        loginId: loginId,
+        ltiId: token.user
+      })
+      /* exclusively treating null value as false, But in really the const `created` won't be null with mysql dialect.
+      This is typescript being smart. See more comments on this in user.service.ts */
+      logger.info(
```

Comment:
Maybe this could be `debug` level; I'm not sure whether we'd want it to be included in a prod log or not.

Reply:
Done